src/components/cl/basic/cl_basic_team.c

```diff
@@ -61,8 +71,37 @@ ucc_status_t ucc_cl_basic_team_create_test(ucc_base_team_t *cl_team)
     ucc_cl_basic_team_t    *team = ucc_derived_of(cl_team, ucc_cl_basic_team_t);
     ucc_cl_basic_context_t *ctx  = UCC_CL_BASIC_TEAM_CTX(team);
     ucc_status_t            status;
+    ucc_base_team_t        *b_team;
+
     status = UCC_TL_CTX_IFACE(ctx->tl_ucp_ctx)
                  ->team.create_test(&team->tl_ucp_team->super);
+    if (UCC_OK == status) {
```

Comment:
This could be some common functionality for CL. Ie, some CLs may want to create multiple TL teams. E.g., hierarchical CL will create multiple different TL teams at differet hierarchy levels.
 It would be beneficial to make some helper routine at CL level which is used during cl team_create_post/test. For example:
we define a struct like {tl_team_ptr **tl_team, params, tl_ctx}. Then during team_create_post CL allocates an array of such structs that will describe ALL the TL teams it needs. Next CL calls smth like: cl_create_tl_teams_post(array_of_descriptors, &request). And from the test fn it calls the corresponding cl_create_tl_teams_test(request). 

In your example the array of descriptiors would have 2 positions: ucp and nccl. @Sergei-Lebedev wdyt ?

Reply:
Agree, it would be helpful